// ngx-rating-stars/esm2015/lib/ngx-rating-stars.component.js
import { Component, Input } from '@angular/core';
export class NgxRatingStarsComponent {
    constructor() {
        this.showRatingInNumericFormat = false;
        this.yellowStar = "/assets/images/" + "yellow-star.jpg";
        this.midStar = "/assets/images/" + "mid-star.jpg";
        this.greyStar = "/assets/images/" + "grey-star.jpg";
        this.ratingArray = new Array;
    }
    ngOnInit() {
        this.buildRatingsStars();
    }
    // builds the array that shows the stars
    buildRatingsStars() {
        let ratingArray = new Array;
        if (this.rating.value == 0) {
            for (let k = 0; k <= 4; k++) {
                ratingArray[k] = this.greyStar;
            }
            this.ratingArray = ratingArray;
        }
        else {
            let rating = this.rating.value;
            let trunc = Math.trunc(rating);
            for (let j = 0; j < trunc + 1; j++) {
                if (rating > j && rating < j + 1) {
                    ratingArray[j] = this.midStar;
                }
                else {
                    if (j != trunc) {
                        ratingArray[j] = this.yellowStar;
                    }
                }
            }
            if (rating == Math.floor(rating)) {
                for (let k = trunc; k <= 4; k++) {
                    ratingArray[k] = this.greyStar;
                }
            }
            else {
                for (let k = trunc + 1; k <= 4; k++) {
                    ratingArray[k] = this.greyStar;
                }
            }
            this.ratingArray = ratingArray;
        }
    }
}
NgxRatingStarsComponent.decorators = [
    { type: Component, args: [{
                selector: 'ngx-rating-stars',
                template: `
        <div class="rating-star" *ngFor="let star of ratingArray"><img [ngStyle]="style.starsStyle" src="{{star}}"></div>
        <div class="rating-values" [ngStyle]="style.ratingStyle"></div>
        <p *ngIf="showRatingInNumericFormat">{{rating?.value | number:'1.1-1'}}</p> 
        <p [ngStyle]="style.countStyle">({{rating?.count}})</p>
    `,
                styles: ["img{height:auto;width:27px}.rating-star,.rating-values,p{display:inline}"]
            },] }
];
NgxRatingStarsComponent.propDecorators = {
    rating: [{ type: Input }],
    style: [{ type: Input }],
    showRatingInNumericFormat: [{ type: Input }]
};
